components/eam/src/physics/crm/ndrop.F90

```diff
@@ -325,8 +325,8 @@ subroutine dropmixnuc( &
    real(r8), intent(in) :: wsub(pcols,pver)    ! subgrid vertical velocity
    real(r8), intent(in) :: cldn(pcols,pver)    ! cloud fraction
    real(r8), intent(in) :: cldo(pcols,pver)    ! cloud fraction on previous time step
-   logical,  intent(in),optional :: do_mmf      ! value insignificant - if variable present, is called in the mmf part.
-   integer, intent(in) :: species_class(:)    
+   integer,  intent(in), optional :: species_class(:) ! needed for MMF
+   logical,  intent(in), optional :: do_mmf           ! if present, is called in the MMF part.
```

Comment:
You removed the part about the value of this flag being insignificant. This is actually super mis-leading...someone could pass `do_mmf=.false.`, and the code behaves the same as if `do_mmf=.true.`. I'd suggest putting a comment in here, or, better yet, fixing the logic below.

Reply:
Changed this logic in recent commits. Decided to keep the "do_mmf" flag and omit the check for "use_MMF", but now "do_mmf" has to be present and true for the modifications to take effect. I also used the do_mmf_local as you suggested.